libs/elasticsearch/langchain_elasticsearch/cache.py

```diff
@@ -60,10 +78,37 @@ def __init__(
             es_api_key: API key to use when connecting to Elasticsearch.
             es_params: Other parameters for the Elasticsearch client.
         """
+
+        self._index_name = index_name
+        self._store_input = store_input
+        self._store_input_params = store_input_params
+        self._metadata = metadata
+        self._setup_connection(
+            es_connection=es_connection,
+            es_url=es_url,
+            es_cloud_id=es_cloud_id,
+            es_user=es_user,
+            es_api_key=es_api_key,
+            es_password=es_password,
+            es_params=es_params,
+        )
+        self._manage_index()
+
+    def _setup_connection(
+        self,
+        *,
+        es_connection: Optional["Elasticsearch"] = None,
```

Comment:
```suggestion
        es_connection: Optional[Elasticsearch] = None,
```
I believe this works since we rely on this class anyways.